apis/kueue/v1beta1/resourceflavor_types.go

```diff
@@ -35,6 +35,11 @@ type ResourceFlavor struct {
 	Spec ResourceFlavorSpec `json:"spec,omitempty"`
 }
 
+// Ref returns the ResourceFlavorReference of a ResourceFlavor
+func (r *ResourceFlavor) Ref() ResourceFlavorReference {
+	return ResourceFlavorReference(r.Name)
```

Comment:
I have mixed feelings about it, because we generally don't add functions to the API package - I didn't see this practiced in k8s. This is harmless, but I'm thinking over time people may want to add more functions here. 
So, I would propose to look for an alternative, maybe pkg/util/refs.go, and some function there?

Reply:
What is the reason that we need to provide such converter function?
Could we directly handle the resource flavor name as typed `ResourceFlavorReference` instead of string?

Reply:
This would work for me too, I think the author's idea was to make the code more concise, which I can buy too

Reply:
I'm not still sure the reason. For example, why not change the typed to ResourceFlavorReference in the following?

https://github.com/kubernetes-sigs/kueue/blob/f1e132bdaad5419c277e16b01acc04b530b4130f/pkg/cache/clusterqueue.go#L605

Reply:
Yeah, this looks like an omission

Reply:
I'm wondering if we can get rid of this converter once we replace all string flavor names with ResourceFlavorReference typed.

Reply:
> I'm not still sure the reason. For example, why not change the typed to ResourceFlavorReference in the following?

That change is indeed in this PR: https://github.com/kaisoz/kueue/blob/4a77f59347fac56daa2260e6e730178f64d7f4c7/pkg/cache/clusterqueue.go#L605

> I'm wondering if we can get rid of this converter once we replace all string flavor names with ResourceFlavorReference typed.

The converter is not needed. My initial approach was to use `ResourceFlavorReference` everywhere. Then I thought that casting the `ResourceFlavor.Name` (a string) to a `ResourceFlavorReference` could be written in a more concise way (it can get lengthy, especially in test code). For example, this:

```
c.resourceFlavors[kueue.ResourceFlavorReference(rf.Name)] = rf
```
could be written like this

```
c.resourceFlavors[rf.Ref()] = rf
```

However, as I discussed with @mimowo on Slack, I was concerned about this approach since I don't see methods being used in API files. That's why I wrote this proposal to get your opinions. My bad for not adding this concern to the PR description 🤦🏻

After hearing your thoughts, I’ll use the cast to ResourceFlavorReference everywhere instead of the converter 😊